src/test/lib/statistics/cardinality_estimator_test.cpp

```diff
@@ -931,11 +942,47 @@ TEST_F(CardinalityEstimatorTest, Union) {
     node_b);
   // clang-format on
 
-  EXPECT_DOUBLE_EQ(estimator.estimate_cardinality(input_lqp), 132.0);
+  const auto left_rowcount = node_a->table_statistics()->row_count;
+  const auto right_rowcount = node_b->table_statistics()->row_count;
+  const auto expected_selectivity = (left_rowcount + right_rowcount) / (left_rowcount);
 
   const auto result_statistics = estimator.estimate_statistics(input_lqp);
 
   ASSERT_EQ(result_statistics->column_statistics.size(), 2);
+  EXPECT_DOUBLE_EQ(result_statistics->row_count,
+                   node_a->table_statistics()->row_count + node_b->table_statistics()->row_count);
+
+  for (auto column_id = ColumnID{0}; column_id < result_statistics->column_statistics.size(); ++column_id) {
+    const auto attribute_statistics_a_a = std::dynamic_pointer_cast<const AttributeStatistics<int32_t>>(
+        node_a->table_statistics()->column_statistics.at(column_id));
+
+    const auto res_column = std::dynamic_pointer_cast<const AttributeStatistics<int32_t>>(
+        result_statistics->column_statistics.at(column_id));
+
+    const auto scaled_histogram = dynamic_cast<const ScaledHistogram<int32_t>*>(res_column->histogram.get());
+
+    EXPECT_EQ(expected_selectivity, expected_selectivity);
+    EXPECT_EQ(ScaledHistogramTestUtils::selectivity(*scaled_histogram), expected_selectivity);
+    EXPECT_EQ(ScaledHistogramTestUtils::referenced_histogram(*scaled_histogram), attribute_statistics_a_a->histogram);
+  }
+}
+
+TEST_F(CardinalityEstimatorTest, UnionWithEmptyInput) {
+  // Test that UnionNode return the table statistics of the non-empty input when the other input is empty
```

Comment:
Please also add plans where the left input is empty and both are empty (can be in the same test case).